src/Control/Carrier/Parse/Simple.hs

```diff
@@ -26,14 +27,14 @@ runParse :: Duration -> ParseC m a -> m a
 runParse timeout (ParseC m) = runReader timeout m
 
 newtype ParseC m a = ParseC (ReaderC Duration m a)
-  deriving (Applicative, Functor, Monad, MonadIO)
+  deriving (Applicative, Functor, Monad, MonadFail, MonadIO)
```

Comment:
This is the big change: because `MonadFailDesugaring` is on by default now, we need more `MonadFail` instances rather than relying on `fail` from `Monad`.

Reply:
I’m kind of appalled to realize that we were ever using `Monad`’s `fail` 😬